### PR TITLE
Add circle config for branch tests

### PR DESCRIPTION
**Pull request by SwampDragons**
need to be able to run branch tests on this repo

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        type: string
+    steps:
+      - checkout
+      - run: GOOS=<< parameters.GOOS >> GOARCH=<<parameters.GOARCH>> go build -ldflags="-s -w -X github.com/hashicorp/packer/version.GitCommit=${CIRCLE_SHA1}" -o ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >> .
+      - run: zip ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>.zip ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - run: rm ./pkg/packer_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./pkg/
```
Comment (sylviamoss) [inline .circleci/config.yml:53]: This is very packer-specific. Plugins don't have the GitCommit commit constant, we will need to implement that.  I also think the zip should have the plugin name.

**Review by sylviamoss — CHANGES_REQUESTED** [review]
This looks good! 
I'm requesting changes because of the persistent part, I think it has to be adapted for plugins. Also, I would add this to our packer-repositories to add it to all of the plugins 🤔  what do you think?

**Comment by sylviamoss** [comment]
Another option is to update the pull request action to also run for branches with something like 
```
on: [push, pull_request]
```

**Comment by nywilken** [comment]
@sylviamoss these are some great suggestions. We merged a PR last week into packer-repositories to push this change to all repos. Would you mind making your suggested changes to the packer-repositories since you have these top of mind so that we can push the final config to all repos at once. I will approve, push, and apply. 

I'm closing this PR as it has been superseded by the packer-repositories change.

**Pull request closed without merging**